presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -190,6 +190,34 @@ public void testFiltersAndProjections()
         assertQuery("SELECT if(orderkey % 2 = 0, quantity_by_linenumber) FROM orders_ex");
     }
 
+    @Test
+    public void testAnalyzeStats()
+    {
+        assertUpdate("ANALYZE region", 5 /* Expected number of rows updated */);
+
+        // Show stats returns the following stats for each column in region table:
+        // column_name | data_size | distinct_values_count | nulls_fraction | row_count | low_value | high_value
+        assertQuery("SHOW STATS FOR region");
+
+        // Create a partitioned table and run analyze on it.
+        String tmpTableName = generateRandomTableName();
+        try {
+            Session writeSession = buildSessionForTableWrite();
+            getQueryRunner().execute(writeSession, String.format("CREATE TABLE %s (name VARCHAR, regionkey BIGINT," +
+                    "nationkey BIGINT) WITH (partitioned_by = ARRAY['regionkey','nationkey'])", tmpTableName));
+            getQueryRunner().execute(writeSession,
+                    String.format("INSERT INTO %s SELECT name, regionkey, nationkey FROM nation", tmpTableName));
+            assertQuery(String.format("SELECT * FROM %s", tmpTableName),
+                    "SELECT name, regionkey, nationkey FROM nation");
+            assertUpdate(String.format("ANALYZE %s", tmpTableName), 25);
+            assertQuery(String.format("SHOW STATS for %s", tmpTableName));
```

Comment:
What does this do? How do you verify the results of the SHOW STATS command?

Reply:
CMIW, assertQuery compares the results of presto and prestissimo. Do I need to explicit check for column values?
https://prestodb.io/docs/current/sql/show-stats.html

Reply:
I'm thinking that you run ANALYZE on Prestissimo and therefore we don't know if it works or not. If it doesn't work, then SHOW STATS will show nothing for both Prestissimo and Presto and the test will pass. That's not what we want, is it?

Reply:
Also, if ANALYZE works but computes stats incorrectly, the tests won't detect that, will it?

Reply:
@mbasmanova : We had been thinking about what would be a more complete test here to cover the situations you outlined.

i) One option is to compare the results of SHOW STATS with values expected in a static list. This list is populated with what we consider correct results. 
ii) The Java side also does Statement object level tests. We don't have that framework in prestissimo testing.

Right now the only guarantee is that the sequence of operations on either side are consistent. But i) would make it more complete.

Reply:
@mbasmanova Aditi is talking about something like this: https://github.com/prestodb/presto/blob/master/presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java#L4426

Reply:
> i) One option is to compare the results of SHOW STATS with values expected in a static list. This list is populated with what we consider correct results.

Sounds reasonable to me. Thanks.